api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -50,11 +50,25 @@ def remove_line_item
           def empty
             spree_authorize! :update, spree_current_order, order_token
 
-            # TODO: we should extract this logic into service and let
-            # developers overwrite it
-            spree_current_order.empty!
+            result = spree_current_order.empty!
 
-            render_serialized_payload { serialized_current_order }
+            if result.success?
+              render_serialized_payload { serialized_current_order }
+            else
+              render_error_payload(result.error)
+            end
+          end
+
+          def destroy
+            spree_authorize! :update, spree_current_order, order_token
+
+            result = spree_current_order.destroy!
```

Comment:
@KacperMekarski let's also add a method `can_be_deleted?` to `Order` model - we should not allow deleting completed orders, only incomplete without finalized payments

Reply:
done!